### PR TITLE
fixing quoteservice metrics exporting

### PR DESCRIPTION
**Pull request by brettmc**
# Changes

metrics exporting was quietly failing due to misconfiguration. Update some packages to pick up the relevant bugfix, and use OTEL_EXPORTER_OTLP_ENDPOINT so that metrics, traces and eventually logs can use it.
In passing, enable some logging which makes future errors more obvious.

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [x] Appropriate documentation updates in the [docs][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/

Fixes: #775

### CODE REVIEW

**Comment by julianocosta89** [comment]
Thanks for taking care of this @brettmc!
Could you add a `CHANGELOG` entry?

**Review by julianocosta89 — COMMENTED** [review]
Tried to build it locally and started getting the following msg in the logs:
```
[2023-03-15T10:50:43.962596+00:00] otel-php.WARNING: tracestate discarded, invalid member:  {"source":"OpenTelemetry\\API\\Trace\\TraceState"} []
```
Is that expected?

**Comment by brettmc** [comment]
@julianocosta89 no it's not expected...but I did turn on some logging so it's possible that it's a problem that's always been there but not noticed. I didn't see this when just calling the service directly (which would make sense, there is no distributed tracing).

To replicate, do I just bring up everything and watch it run?

File: CHANGELOG.md [hunk 1]
```diff
@@ -271,3 +271,7 @@ significant modifications will be credited to OpenTelemetry Authors.
   ([#764](https://github.com/open-telemetry/opentelemetry-demo/pull/764))
 * [chore] align memory limits with Helm chart
   ([#781](https://github.com/open-telemetry/opentelemetry-demo/pull/781))
+* [bug] fixing quoteservice metrics exporting (PHP)
+  ([#793)(https://github.com/open-telemetry/opentelemetry-demo/pull/793))
+D
+  ([#793]())
```
Comment (julianocosta89) [inline CHANGELOG.md:277]:
```suggestion
* [bug] fixing quoteservice metrics exporting (PHP)
  ([#793)(https://github.com/open-telemetry/opentelemetry-demo/pull/793))
```

**Comment by julianocosta89** [comment]
> @julianocosta89 no it's not expected...but I did turn on some logging so it's possible that it's a problem that's always been there but not noticed. I didn't see this when just calling the service directly (which would make sense, there is no distributed tracing).
> 
> To replicate, do I just bring up everything and watch it run?

You can run:
```
docker compose build quoteservice --no-cache

docker compose up
```

Once everything is running, you can get the logs from quoteservice:
```
docker logs quoteservice
```

**Comment by brettmc** [comment]
upstream bugfix: https://github.com/open-telemetry/opentelemetry-php/pull/936

**Comment by brettmc** [comment]
ok, patched in a new version of one of our packages, the warning message is gone, and we have one less bug in our code. I think this is ready now.

**Review by julianocosta89 — APPROVED** [review]
Awesome!

**Review by mviitane — APPROVED** [review]

**Pull request merged**